openstack/networking/v1/eips/requests.go

```diff
@@ -2,8 +2,48 @@ package eips
 
 import (
 	"github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/pagination"
 )
 
+// ListOptsBuilder allows extensions to add additional parameters to the
+// List request.
+type ListOptsBuilder interface {
+	ToEipListQuery() (string, error)
+}
+
+// ListOpts filters the results returned by the List() function.
+type ListOpts struct {
+	// Marker specifies the start resource ID of pagination query
+	Marker string `q:"marker"`
+
+	// Limit specifies the number of records returned on each page.
+	Limit int `q:"limit"`
+}
+
+// ToEipListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToEipListQuery() (string, error) {
+	q, err := golangsdk.BuildQueryString(opts)
+	if err != nil {
+		return "", err
+	}
+	return q.String(), err
+}
+
+// List instructs OpenStack to provide a list of flavors.
+func List(client *golangsdk.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := rootURL(client)
+	if opts != nil {
+		query, err := opts.ToEipListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return EipPage{pagination.SinglePageBase(r)}
```

Comment:
You can use `MarkerPageBase` for marker-based pagination

Reply:
done